pageserver/src/tenant/timeline.rs

```diff
@@ -5061,6 +5092,19 @@ impl Timeline {
                 }
             }
 
+            // 3.5 Is there a valid lease that requires us to keep this layer?
+            if let Some(lsn) = &min_lsn_with_valid_lease {
+                if &l.get_lsn_range().start <= lsn {
```

Comment:
`layer.get_lsn_range().start` <= `min_lsn_with_valid_lease` <= all other lsn with valid leases.

Reply:
The comparison should actually be against `max_lsn_with_valid_lease` since we keep a layer if the layer start is <= any of the lease.